internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go

```diff
@@ -934,15 +940,24 @@ func (c *ClusterManager) buildObservatoriumSSOSecretResource() *k8sCoreV1.Secret
 	observabilityConfig := c.ObservabilityConfiguration
 	stringDataMap := map[string]string{
 		"authType":               observatorium.AuthTypeSso,
-		"gateway":                observabilityConfig.RedHatSsoGatewayUrl,
 		"tenant":                 observabilityConfig.RedHatSsoTenant,
-		"redHatSsoAuthServerUrl": observabilityConfig.RedHatSsoAuthServerUrl,
-		"redHatSsoRealm":         observabilityConfig.RedHatSsoRealm,
-		"metricsClientId":        observabilityConfig.MetricsClientId,
-		"metricsSecret":          observabilityConfig.MetricsSecret,
-		"logsClientId":           observabilityConfig.LogsClientId,
-		"logsSecret":             observabilityConfig.LogsSecret,
+		"gateway":                "",
+		"redHatSsoAuthServerUrl": "",
+		"redHatSsoRealm":         "",
+		"metricsClientId":        "",
+		"metricsSecret":          "",
+	}
+
+	if observabilityConfig.DataPlaneObservabilityConfig.Enabled {
+		stringDataMap["gateway"] = observabilityConfig.DataPlaneObservabilityConfig.RemoteWriteConfiguration.RemoteWriteURL
+		if observabilityConfig.DataPlaneObservabilityConfig.HasStaticOIDCConfiguration() {
```

Comment:
This is possibly not the best approach. We should have some type that simply gets the desired OIDC configuration elements without asking on the specific type of configuration.
Let's leave it like this for now but let's try to improve it in a future PR.